components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1220,12 +1220,13 @@ private String populateOauthParameters(OAuth2Parameters params, OAuthMessage oAu
         } else {
             params.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
         }
-        if (StringUtils.isNotBlank(oauthRequest.getParam(ACR_VALUES)) && !"null".equals(oauthRequest.getParam
-                (ACR_VALUES))) {
-            String[] acrValues = oauthRequest.getParam(ACR_VALUES).split(" ");
-            LinkedHashSet<String> list = new LinkedHashSet<>();
-            list.addAll(Arrays.asList(acrValues));
-            params.setACRValues(list);
+        if (StringUtils.isNotBlank(oauthRequest.getParam(ACR_VALUES_ATTRIBUTE)) && !"null".equals(oauthRequest.getParam
```

Comment:
We need to fix this "null" being passed. IMO String "null" should be a valid value. Better to create a new issue.